internal/adapter/lsp/document.go

```diff
@@ -229,10 +246,47 @@ func (d *document) DocumentLinks() ([]documentLink, error) {
 			})
 		}
 
+		if insideFenced {
+			if fencedEndRegex.FindStringIndex(line) != nil &&
+			lines[currentCodeBlockStart][:3] == line[:3] {
+				// Fenced code block ending with this line
+				insideFenced = false
+				currentCodeBlockStart = -1
+			}
+			// Within fenced code block, skip link checks
+			continue
+		} else if insideIndented {
+			if indentedRegex.FindStringIndex(line) == nil && len(line) > 0 {
+				// No longer indented, will process links
+				insideIndented = false
+				currentCodeBlockStart = -1
+			} else {
+				// Still indented, skip link checks
+				continue
+			}
+		} else {
+			// Check whether we are in fenced / indented code check
+			if fencedStartRegex.FindStringIndex(line) != nil {
+				insideFenced = true
+				currentCodeBlockStart = lineIndex
+				continue
+			} else if indentedRegex.FindStringIndex(line) != nil &&
+			(lineIndex > 0 && len(lines[lineIndex-1]) == 0 || lineIndex == 0) {
+				// Indented code block starts on this line
+				insideIndented = true
+				currentCodeBlockStart = lineIndex
+				continue
+			}
+		}
```

Comment:
Does it make sense to place this block above the declaration of appendLink()? Just to shave down some execution time?

As otherwise, while on a line in a code block for example, appendLink() will be redeclared with data that will never be used. That line will be fully skipped anyway, meaning that the newly declared appendLink() won't even be called?